enhancements/sig-architecture/30-clusterset-override/README.md

```diff
@@ -0,0 +1,239 @@
+   resourceNames: ["region"]
+   verbs: ["create"]
+ - apiGroups: ["cluster.open-cluster-management.io"]
+   resources: ["managedclustersets"]
+   resourceNames: ["apac"]
+   verbs: ["*"]
+```
+ 
+3. Create a managedClusterSet to select all cluster in apac
+```yaml
+apiVersion: cluster.open-cluster-management.io/v1alpha1
+kind: ManagedClusterSet
+metadata:
+name: apac
+spec:
+exclusiveKey: region
+```
+ 
+This managedClusterSet includes all clusters with label `region: apac`, so I could use this managedClusterSet to select all clusters in apac, then apply a certain configuration to these clusters.
+ 
+### I want to create a managedClusterSet for Disaster Recovery purposes, the managedClusterSet should select two or more clusters in different regions.
+1. I need to manually select one cluster in each region: `apac`,`us`,`europe`, and set label: `clusterAttribute=backup` for these clusters
+2. Get the following permission to create a managedClusterSet.
+```yaml
+kind: ClusterRole
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+ name: clustersetrole
+rules:
+ - apiGroups: ["cluster.open-cluster-management.io"]
+   resources: ["managedclustersets/exclusiveKey"]
+   resourceNames: ["clusterAttribute"]
+   verbs: ["create"]
+ - apiGroups: ["cluster.open-cluster-management.io"]
+   resources: ["managedclustersets"]
+   resourceNames: ["backup"]
+   verbs: ["*"]
+```
+ 
+3. Create a managedClusterSet to select
+```yaml
+apiVersion: cluster.open-cluster-management.io/v1alpha1
+kind: ManagedClusterSet
+metadata:
+name: backup
+spec:
+exclusiveKey: clusterAttribute
+```
+ 
+### I want to create a managedClusterSet to group clusters based on the middleware enabled on these clusters, so I could run special applications in these clusters.
+1. For all clusters which enabled middleware, I need to set label: `middlewareAttribute=middlewareenabled`
+2. Get the following permission to create a managedClusterSet.
+```yaml
+kind: ClusterRole
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+ name: clustersetrole
+rules:
+ - apiGroups: ["cluster.open-cluster-management.io"]
+   resources: ["managedclustersets/exclusiveKey"]
+   resourceNames: ["middlewareAttribute"]
+   verbs: ["create"]
+ - apiGroups: ["cluster.open-cluster-management.io"]
+   resources: ["managedclustersets"]
+   resourceNames: ["middlewareenabled"]
+   verbs: ["*"]
+```
+ 
+3. Create a managedClusterSet to select all clusters which enabled middleware.
+```yaml
+apiVersion: cluster.open-cluster-management.io/v1alpha1
+kind: ManagedClusterSet
+metadata:
+name: middlewareenabled
+spec:
+exclusiveKey: middlewareAttribute
+```
+ 
+### Risks and Mitigation
+Currently, if I want to create a managedClusterSet, I must get permissions about two kinds of resources, `managedclustersets/exclusiveKey` and `managedclustersets`. When there are multiple resources for both `managedclustersets/exclusiveKey` and `managedclustersets`, it may cause permission leak.
```

Comment:
There is another risk. Since permission check will no longer be done when adding a cluster into a clusterset. User may add a cluster into a clusterset incidentally by adding a label to cluster, which is the exclusiveKey of that clusterset.